### PR TITLE
DBT Performance Logging and Analysis with Graphviz

### PR DESCRIPTION
**Pull request by mescanne**
Series of enhancements:
- DBT package for logging on start of run and end of run
- Python script to inspect and extract peformance JSON for individual runs
- Python script to convert performance JSON to a graphviz DOT graph
- Instrumentation of homeloan use case
- Documentation on using this package

### CODE REVIEW

File: common_components/monitoring/json_to_dot.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    return name.replace('.','__').replace("-", "__")
+
+
+def trim_names(nodes):
+    """ Remove common prefixes from all graph names (DBT can be verbose!) """
+    names = [n['name'] for n in nodes.values()]
+    prefix = os.path.commonprefix(names)
+    if prefix:
+        for node in nodes.values():
+            node['name'] = node['name'].removeprefix(prefix)
```
Comment (mtronci) [inline common_components/monitoring/json_to_dot.py:48]: the removeprefix method does not work with anything earlier than 3.9.
Is it worth using a syntax compatible with other older versions here? Something like the below:
`node['name'] = node['name'][len(prefix):]`

**Review by mtronci — COMMENTED** [review]
Hi @mescanne , the PR looks great.
Just one minor comment on the `json_to_dot.py` script, appreciate if you could take a look before merging.

File: common_components/monitoring/json_to_dot.py (hunk 1 above)
Comment (mescanne) [reply common_components/monitoring/json_to_dot.py:48]: Replaced removeprefix as suggested -- good idea.

On the subject of google-cloud-bigquery dependency -- I just tried it out and all that's required is install google-cloud-dependency, but it also brings in a lot of other dependencies.

I think we can -
 - have a fully specified requirements.txt (I'd rather not..)
 - have just google-cloud-bigquery which will implicitly pull in dependencies (we'll need an exact version)
 - document in the README and in the script that google-cloud-bigquery needs to be installed.
 - capture the ModuleNotFoundError and document in the script that google-cloud-bigquery needs to be installed

What do you think about just doing the last two?

Comment (mtronci) [reply common_components/monitoring/json_to_dot.py:48]: yes, doing the last two sounds great

Comment (mescanne) [reply common_components/monitoring/json_to_dot.py]: Code updated with changes. Let me know.

**Review by mtronci — APPROVED** [review]
Thanks @mescanne , approved!

**Pull request merged**